smithy-base-plugin/src/main/java/software/amazon/smithy/gradle/internal/CliDependencyResolver.java

```diff
@@ -83,24 +84,27 @@ private static String getCliVersion(Project project, SourceSet sourceSet) {
     /**
      * Check if there's a dependency on smithy-model somewhere, and assume that version.
      *
-     * @param project configuration to search for CLI version
-     * @param sourceSet SourceSet to get runtime configuration for
+     * @param configurations Configuration contain to use to search for configurations
      *
      * @return version of cli available in configuration
      */
-    public static String detectCliVersionInRuntimeDependencies(Project project, SourceSet sourceSet) {
-        Configuration runtimeClasspath = project.getConfigurations().getByName(
-                sourceSet.getRuntimeClasspathConfigurationName());
+    public static String detectCliVersionInRuntimeDependencies(ConfigurationContainer configurations) {
+        // If the runtimeClasspath configuration does not exist we cannot scan runtime deps
+        if (configurations.findByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME) == null) {
```

Comment:
I wonder if we should allow users to configure what classpath to use for discovery through the extension?

Reply:
This scanning is what allows use to pull the version of the CLI to use from the configured dependency. For example if a customer sets the following dependency:
```
dependencies {
    implementation("software.amazon.smithy:smithy-model:1.40.0")
}
```
We try to ensure that we resolve the CLI version to `1.40.0` as well. In order to do that we need to scan the `runtimeClasspath` configuration for the smithy-model dependency. However, the `runtimeClasspath` configuration only exists if the `Java` base plugin has been applied.